src/ssh/cockpitsshtransport.c

```diff
@@ -538,11 +514,13 @@ cockpit_ssh_transport_set_property (GObject *obj,
     case PROP_KNOWN_HOSTS:
       self->knownhosts_file = g_value_dup_string (value);
       break;
-    case PROP_CREDS:
-      self->creds = g_value_dup_boxed (value);
+    case PROP_USER:
+      self->user = g_value_dup_string (value);
       break;
-    case PROP_AGENT:
-      self->agent = g_value_dup_object (value);
+    case PROP_PASSWORD:
+      password = g_value_get_string (value);
+      if (password)
+        self->password = g_bytes_new_take (g_strdup (password), strlen (password));
```

Comment:
I wonder why none of the previous values get freed here (in the existing code too). Can/do we assume that we only ever set the properties once?

Update: yes, we apparently use `CONSTRUCT_ONLY` for those, so nevermind.